### PR TITLE
Feature 3522 lbds prototype

### LINKED ISSUE #3522: Prototype integration of LBDS in SORMAS app [8]
### Situation Description
An Low-Bandwidth-Data-Synchronization is being developed to send data from mobile app to the server based on SMS.
LBDS will be running as an app on it's own on the Android device and receive and send intents to do the actual communication.
API to be used: https://github.com/crowdcode-de/lbds-android-messaging

### Prerequisits

@gorefest We will need the following things to implement this and try whether it's working

- [x] The lbds-android-messaging package via maven central

Question: Is there any way we can try whether this is working, once we start to implement it? Or should we just add it and you will try to make it work?

### Feature Description

- [x] Create a class that receives LbdsResponseIntents
- [ ] Update the https://github.com/crowdcode-de/lbds-android-messaging project with the name of the actual class that will receive the intents from LBDS. Can be commited by @MartinWahnschaffeSymeda
  **still a typo in class name `LbdsRecevierComponent`. This has to be changed synchronously in the lbds service app**
- [x] Include a button in the dev mode view of the app that allows sending all ~~the first~~ unsynchronized cases
- [x] Use the CaseDataDto and PersonDto, but only send the required fields of the case and person. Leave all other fields empty. Make sure the generated JSON does not create empty attribute entries. 
- [x] ~~To identify which persons have to be sent, check for persons with "modified" set to true, that at the same time are associated to a case (case.person_id). Among these persons identify the ones that have changes in the essential data (as listed below). You can identify the changes by comparing the modified version of the person with the unmodified snapshot that can be retrieved with AbstractAdoDao.querySnapshotByUuid (when no snapshot is returned, it means that the person has not been sent to the server before).~~
- [x] ~~To identify which cases need to be sent, check for cases with "modified" set to true. Then also compare the essential fields of the case, as done for the person OR add a new date field that contains the info when the data was sent last via LBDS~~
- [x] ~~While the persons and cases are being sent via LBDS, we need to somehow store which data is currently being send. Once the data has been sent, we have to either update the existing snapshot with the values of the essential fields that have been sent OR create a snapshot based on an empty case and fill it with the sent values. There is currently no mechanism for this, so this will be quite difficult.~~
  **to be solved
  **sent requests are persisted in the LBDS service app and processed upon receiving a response**
- [ ] ~~Process http response from LBDS and update snapshots / modified flag accordingly~~
  ~~**modified flag is reset, snapshot has to be written**~~
- [x] Show LBDS buttons only if the LBDS app is installed
- [x] Make an additional SORMAS-LBDS URL configurable for debugging purposes
- [x] Use encryption to store LBDS credentials and secrets
- [x] Document usage and installation + debug setup
- [x] Create a table to persist times of sent and received objects. Table name: LbdsSync, Columns: Uuid, lbdsSent, lbdsReceived. This table provides the criteria whether or not to send an object via LBDS and replaces the criteria stated above.
- [x] To be send via LBDS are: new Cases for which no LBDS response has been received, new Persons of these cases for which no LBDS response has been received.
- [x] Show a notification when LBDS messages for cases or persons  are sent or received

Difficulties related to the whole process

1. LBDS is async (minutes/hours), this means that it can easily happen that a user changes the data while it's still being sent
2. We are only sending a part of the data, so after it's sent, we only know a partial snapshot of the truth on the server
3. We can't pull data before sending, which means that when sending changed data this may conflict with changed data on the server. Currently this is denied based on the change date (TOO_OLD) -> for now only send NEW cases via LBDS

-> @syntakker will create process charts to have a better discussion basis for this

The following fields should be sufficient:

* Person: uuid, first name, last name, sex
* Case: uuid, person reference (uuid), disease, report data, reporting user (uuid), region, district, facility type, health facility (uuid)

### Limitations of the Prototype

- Prototype only sends new cases, not changes of existing ones
- Sending must be done via the hidden developer area in the settings
- Persons and cases must be sent separately
- A bit more data sent than necessary -> #5891

### PR DESCRIPTION
**Pull request by syntakker**
Fixes #3522

### CODE REVIEW

File: sormas-app/app/src/main/java/de/symeda/sormas/app/settings/SettingsFragment.java [hunk 1]
```diff
@@ -88,12 +93,22 @@ public View onCreateView(LayoutInflater inflater, @Nullable ViewGroup container,
 		binding.resynchronizeData.setOnClickListener(v -> repullData());
 		binding.showSyncLog.setOnClickListener(v -> openSyncLog());
 		binding.logout.setOnClickListener(v -> logout());
+		binding.kexLbds.setOnClickListener(v -> kexLbds());
+		binding.syncPersonLbds.setOnClickListener(v -> LbdsIntentSender.sendPersonsLbds(getContext()));
+		binding.syncCaseLbds.setOnClickListener(v -> LbdsIntentSender.sendCasesLbds(getContext()));
```
Comment (MartinWahnschaffe) [inline sormas-app/app/src/main/java/de/symeda/sormas/app/settings/SettingsFragment.java:98]: synCases should send the new cases including the related persons. There should not be a separate button for sending persons

File: sormas-app/keystore.properties.example [hunk 2]
```diff
@@ -1,5 +0,0 @@
-# on linux use / on windows \\ as path seperator
```
Comment (MartinWahnschaffe) [inline sormas-app/keystore.properties.example:1]: This example file should not be deleted

File: sormas-rest/src/main/java/de/symeda/sormas/rest/PersonResource.java [hunk 3]
```diff
@@ -53,7 +53,7 @@
 @Consumes(MediaType.APPLICATION_JSON + "; charset=UTF-8")
 @RolesAllowed({
 	"USER",
-	"REST_USER" })
+	"REST_USER", "ADMIN" })
```
Comment (MartinWahnschaffe) [inline sormas-rest/src/main/java/de/symeda/sormas/rest/PersonResource.java:56]: Should not be changed

File: sormas-api/src/main/java/de/symeda/sormas/api/utils/pseudonymization/DtoPseudonymizer.java [hunk 4]
```diff
@@ -107,7 +107,7 @@ public void pseudonymize(DTO d) {
 		List<Field> embeddedFields = getEmbeddedFields(type, isInJurisdiction);
 
 		for (Field pseudonymizedField : pseudonymizableFields) {
-			if (!getFieldAccessCheckers(isInJurisdiction).isAccessible(pseudonymizedField, pseudonymizeMandatoryFields) || dto.isPseudonymized()) {
+			if (!getFieldAccessCheckers(isInJurisdiction).isAccessible(pseudonymizedField, pseudonymizeMandatoryFields) || (dto != null && dto.isPseudonymized())) {
```
Comment (MartinWahnschaffe) [inline sormas-api/src/main/java/de/symeda/sormas/api/utils/pseudonymization/DtoPseudonymizer.java:110]: Should not be part of this PR

File: sormas-app/app/src/main/java/de/symeda/sormas/app/LbdsRecevierComponent.java [hunk 5]
```diff
@@ -0,0 +1,234 @@
...
+	protected void onHandleIntent(@Nullable Intent intent) {
+		Log.i("SORMAS_LBDS", "==========================");
+		final String intentType = intent.getStringExtra(Constants.INTENT_TYPE);
+		Log.i("SORMAS_LBDS", "Received LBDS intent: " + intentType);
+		if (intentType != null && !intentType.trim().isEmpty()) {
+			IntentType type = IntentType.valueOf(intentType);
+			switch (type) {
+			case KEX_TO_LBDS_INTENT:
+				final String publicKey = intent.getStringExtra(Constants.SORMAS_KEY);
+				final PublicKey key = KeySerializationUtil.deserializePublicKey(publicKey);
```
Comment (MartinWahnschaffe) [inline sormas-app/app/src/main/java/de/symeda/sormas/app/LbdsRecevierComponent.java:65]: I guess this is the reply coming from LBDS that the key was received. Anything to do with this?

File: sormas-app/app/src/main/java/de/symeda/sormas/app/LbdsRecevierComponent.java [hunk 6]
```diff
@@ -0,0 +1,234 @@
...
+
+				Log.i("SORMAS_LBDS", "Sormas public key: " + kexToSormasIntent.getSormasKey());
+
+				PublicKey lbdsKey = kexToSormasIntent.getLbdsKey();
+				ConfigProvider.setLbdsServicePublicKey(lbdsKey);
+				Log.i("SORMAS_LBDS", "Lbds public key: " + lbdsKey);
+
+				String aesSecret = kexToSormasIntent.getAesSecret(ConfigProvider.getLbdsSormasPrivateKey());
+				ConfigProvider.setLbdsAesSecret(aesSecret);
+				Log.i("SORMAS_LBDS", "Lbds AES secret: " + Base64.encodeToString(aesSecret.getBytes(StandardCharsets.UTF_8), Base64.DEFAULT));
```
Comment (MartinWahnschaffe) [inline sormas-app/app/src/main/java/de/symeda/sormas/app/LbdsRecevierComponent.java:99]: We should not log the secret

File: sormas-app/app/src/main/java/de/symeda/sormas/app/LbdsRecevierComponent.java [hunk 7]
```diff
@@ -0,0 +1,234 @@
...
+		Log.i("SORMAS_LBDS", "Received LBDS intent: " + intentType);
+		if (intentType != null && !intentType.trim().isEmpty()) {
+			IntentType type = IntentType.valueOf(intentType);
+			switch (type) {
+			case KEX_TO_LBDS_INTENT:
+				final String publicKey = intent.getStringExtra(Constants.SORMAS_KEY);
+				final PublicKey key = KeySerializationUtil.deserializePublicKey(publicKey);
+
+				break;
+			case HTTP_RESPONSE_INTENT:
```
Comment (MartinWahnschaffe) [inline sormas-app/app/src/main/java/de/symeda/sormas/app/LbdsRecevierComponent.java:68]: Please add a short comment to each case, to explain what it is about

File: sormas-app/app/src/main/java/de/symeda/sormas/app/LbdsRecevierComponent.java [hunk 8]
```diff
@@ -0,0 +1,234 @@
...
+			Person snapshot = personDao.querySnapshotByUuid(uuid);
+			if (snapshot != null) {
+				Log.i("SORMAS_LBDS", "Person " + uuid + " has a snapshot, skip.");
+				continue;
+			}
+			try {
+				if (!LbdsDtoHelper.isModifiedLbds(person, personDto, true)) {
+					personDao.accept(person);
+				} else {
+					// TODO: create snapshot
```
Comment (MartinWahnschaffe) [inline sormas-app/app/src/main/java/de/symeda/sormas/app/LbdsRecevierComponent.java:157]: This should be implemented...

File: sormas-app/app/src/main/java/de/symeda/sormas/app/LbdsRecevierComponent.java [hunk 9]
```diff
@@ -0,0 +1,234 @@
...
+			Case snapshot = caseDao.querySnapshotByUuid(uuid);
+			if (snapshot != null) {
+				Log.i("SORMAS_LBDS", "Case " + uuid + " has a snapshot, skip.");
+				continue;
+			}
+			try {
+				if (!LbdsDtoHelper.isModifiedLbds(caze, caseDataDto, true)) {
+					caseDao.accept(caze);
+				} else {
+					// TODO: create snapshot
```
Comment (MartinWahnschaffe) [inline sormas-app/app/src/main/java/de/symeda/sormas/app/LbdsRecevierComponent.java:213]: To be implemented...

File: sormas-app/app/src/main/java/de/symeda/sormas/app/lbds/LbdsDtoHelper.java [hunk 10]
```diff
@@ -0,0 +1,206 @@
...
+	public static boolean isModifiedLbds(Case caze, CaseDataDto caseDataDto, boolean checkNonLbdsProperties)
+		throws IntrospectionException, InvocationTargetException, IllegalAccessException {
+		return isModifiedLbds(caze, caseDataDto, new CaseDtoHelper(), checkNonLbdsProperties);
+	}
+
+	private static <ADO extends AbstractDomainObject, DTO extends EntityDto> boolean isModifiedLbds(
+		ADO ado,
+		DTO dto,
+		AdoDtoHelper<ADO, DTO> adoDtoHelper,
+		boolean checkNonLbdsProperties)
```
Comment (MartinWahnschaffe) [inline sormas-app/app/src/main/java/de/symeda/sormas/app/lbds/LbdsDtoHelper.java:142]: WHen is this needed? Seem to contradict the purpose of the method.

**Review by MartinWahnschaffe — CHANGES_REQUESTED** [review]

File: sormas-api/src/main/java/de/symeda/sormas/api/utils/pseudonymization/DtoPseudonymizer.java (hunk 4 above)
Comment (syntakker) [reply sormas-api/src/main/java/de/symeda/sormas/api/utils/pseudonymization/DtoPseudonymizer.java:110]: I will revert this.

File: sormas-api/src/main/java/de/symeda/sormas/api/utils/pseudonymization/PseudonymizableDto.java [hunk 11]
```diff
@@ -20,6 +20,8 @@
 
 	private static final long serialVersionUID = 4181307802683421947L;
 
+	public static final String PSEUDONYMIZED = "pseudonymized";
```
[the remark this answers is not in the document]
Comment (syntakker) [reply sormas-api/src/main/java/de/symeda/sormas/api/utils/pseudonymization/PseudonymizableDto.java:23]: This is referenced in `LbdsDtoHelper.`, see your comment https://github.com/hzi-braunschweig/SORMAS-Project/pull/6001/files#r661263725 about the String `"ownershipHandedOver"`.

File: sormas-app/app/build.gradle [hunk 12]
```diff
@@ -33,7 +33,7 @@ android {
     }
     buildTypes {
         debug {
-            applicationIdSuffix ".debug"
+            // applicationIdSuffix ".debug"
```
[the remark this answers is not in the document]
Comment (syntakker) [reply sormas-app/app/build.gradle:36]: We have to find a solution here. The ".debug" suffix changes the app package, hence the Receiver Component is not available to the LBDS service app in the debug build.

File: sormas-app/app/src/main/java/de/symeda/sormas/app/LbdsRecevierComponent.java (hunk 7 above)
Comment (syntakker) [reply sormas-app/app/src/main/java/de/symeda/sormas/app/LbdsRecevierComponent.java:68]: I will add comments here.

File: sormas-app/app/src/main/java/de/symeda/sormas/app/LbdsRecevierComponent.java (hunk 6 above)
Comment (syntakker) [reply sormas-app/app/src/main/java/de/symeda/sormas/app/LbdsRecevierComponent.java:99]: Only too true... O_o

File: sormas-app/app/src/main/java/de/symeda/sormas/app/lbds/LbdsDtoHelper.java [hunk 13]
```diff
@@ -0,0 +1,206 @@
...
+		CaseDataDto.NOT_A_CASE_REASON_OTHER,
+		CaseDataDto.NOT_A_CASE_REASON_PHYSICIAN_INFORMATION,
+		CaseDataDto.SHARED_TO_COUNTRY,
+		CaseDataDto.QUARANTINE_ORDERED_VERBALLY,
+		CaseDataDto.QUARANTINE_ORDERED_OFFICIAL_DOCUMENT,
+		CaseDataDto.QUARANTINE_EXTENDED,
+		CaseDataDto.QUARANTINE_REDUCED,
+		CaseDataDto.QUARANTINE_OFFICIAL_ORDER_SENT,
+		CaseDataDto.OVERWRITE_FOLLOW_UP_UNTIL,
+		"ownershipHandedOver");
```
[the remark this answers is not in the document]
Comment (syntakker) [reply sormas-app/app/src/main/java/de/symeda/sormas/app/lbds/LbdsDtoHelper.java:88]: Agree. Same applies to `"enrolledInExternalJournal"`, `"creationVersion"` and `"class"`.

File: sormas-app/app/src/main/java/de/symeda/sormas/app/lbds/LbdsDtoHelper.java (hunk 10 above)
Comment (syntakker) [reply sormas-app/app/src/main/java/de/symeda/sormas/app/lbds/LbdsDtoHelper.java:142]: This is used in two places:
1. to determine if an Ado is suitable to be sent via LBDS (`checkNonLbdsProperties == false`, i.e. LBDS properties have been modified)
2. to check whether a Dto as sent via LBDS is equal to a local Ado, checked on receiving an LBDS response (`checkNonLbdsProperties == true`, the Ado has no extra properties which have not been sent to the SORMAS server)

File: sormas-app/keystore.properties.example (hunk 2 above)
Comment (syntakker) [reply sormas-app/keystore.properties.example:1]: Deleted by mistake, I will restore the file.

File: sormas-rest/src/main/java/de/symeda/sormas/rest/PersonResource.java (hunk 3 above)
Comment (syntakker) [reply sormas-rest/src/main/java/de/symeda/sormas/rest/PersonResource.java:56]: I will revert this.

File: sormas-app/app/src/main/java/de/symeda/sormas/app/backend/caze/CaseDtoHelper.java [hunk 14]
```diff
@@ -86,7 +86,7 @@
 	}
 
 	@Override
-	protected Call<List<PushResult>> pushAll(List<CaseDataDto> caseDataDtos) throws NoConnectionException {
+	public Call<List<PushResult>> pushAll(List<CaseDataDto> caseDataDtos) throws NoConnectionException {
```
Comment (syntakker) [inline sormas-app/app/src/main/java/de/symeda/sormas/app/backend/caze/CaseDtoHelper.java:89]: Changing the visibility here is obsolete, will be reverted.

File: sormas-app/app/src/main/java/de/symeda/sormas/app/backend/person/PersonDtoHelper.java [hunk 15]
```diff
@@ -62,7 +62,7 @@
 	}
 
 	@Override
-	protected Call<List<PushResult>> pushAll(List<PersonDto> personDtos) throws NoConnectionException {
+	public Call<List<PushResult>> pushAll(List<PersonDto> personDtos) throws NoConnectionException {
```
Comment (syntakker) [inline sormas-app/app/src/main/java/de/symeda/sormas/app/backend/person/PersonDtoHelper.java:65]: Visibility change obsolete, as in `CaseDtoHelper`.

File: sormas-app/app/src/main/java/de/symeda/sormas/app/LbdsRecevierComponent.java (hunk 5 above)
Comment (MartinWahnschaffe) [reply sormas-app/app/src/main/java/de/symeda/sormas/app/LbdsRecevierComponent.java:65]: Can be removed

File: sormas-app/app/src/main/java/de/symeda/sormas/app/lbds/LbdsIntentSender.java [hunk 16]
```diff
@@ -0,0 +1,193 @@
...
+		}
+		Log.i("SORMAS_LBDS", "==========================");
+	}
+
+	public static void sendPersonsLbds(Context context) {
+
+		Log.i("SORMAS_LBDS", "==========================");
+		Log.i("SORMAS_LBDS", "Sync Persons LBDS");
+		PersonDao personDao = DatabaseHelper.getPersonDao();
+		List<Person> modifiedPersons = personDao.getModifiedEntities();
```
Comment (MartinWahnschaffe) [inline sormas-app/app/src/main/java/de/symeda/sormas/app/lbds/LbdsIntentSender.java:66]: Don't send all new persons. Instead get the modified cases and send only the new persons of those cases.

File: sormas-app/app/src/main/java/de/symeda/sormas/app/lbds/LbdsIntentSender.java [hunk 17]
```diff
@@ -0,0 +1,193 @@
...
+			Log.i("SORMAS_LBDS", "send SORMAS public key: " + KeySerializationUtil.serializePublicKey(lbdsSormasPublicKey));
+			LbdsPropagateKexToLbdsIntent kexToLbdsIntent = new LbdsPropagateKexToLbdsIntent(lbdsSormasPublicKey);
+			ContextCompat.startForegroundService(context, kexToLbdsIntent);
+		} catch (Exception e) {
+			e.printStackTrace();
+		}
+		Log.i("SORMAS_LBDS", "==========================");
+	}
+
+	public static void sendPersonsLbds(Context context) {
```
Comment (MartinWahnschaffe) [inline sormas-app/app/src/main/java/de/symeda/sormas/app/lbds/LbdsIntentSender.java:61]: sendNewCasePersons

File: sormas-app/app/src/main/java/de/symeda/sormas/app/lbds/LbdsIntentSender.java [hunk 18]
```diff
@@ -0,0 +1,193 @@
...
+			Log.i("SORMAS_LBDS", "Send persons: " + payload);
+			HttpMethod method = new HttpMethod(HttpMethod.MethodType.POST, getLbdsUrl("persons/push"), headers, payload);
+			sendLbdsRequest(context, method);
+		} else {
+			Log.i("SORMAS_LBDS", "Nothing to send.");
+		}
+		Log.i("SORMAS_LBDS", "==========================");
+	}
+
+	public static void sendCasesLbds(Context context) {
```
Comment (MartinWahnschaffe) [inline sormas-app/app/src/main/java/de/symeda/sormas/app/lbds/LbdsIntentSender.java:98]: sendNewCases

**Review by MartinWahnschaffe — CHANGES_REQUESTED** [review]

File: sormas-app/app/src/main/java/de/symeda/sormas/app/lbds/LbdsIntentSender.java [hunk 19]
```diff
@@ -0,0 +1,193 @@
...
+	}
+
+	private static String getLbdsUrl(String path) {
+		String lbdsUrl = ConfigProvider.getServerLbdsDebugUrl();
+		if (lbdsUrl == null || lbdsUrl.isEmpty() || StringUtils.isBlank(lbdsUrl)) {
+			lbdsUrl = ConfigProvider.getServerRestUrl();
+		}
+		String separator = lbdsUrl.endsWith("/") ? "" : "/";
+		return lbdsUrl + separator + path;
+	}
```
Comment (syntakker) [inline sormas-app/app/src/main/java/de/symeda/sormas/app/lbds/LbdsIntentSender.java:192]: URL + Path should be extracted from the `Call` generated by `RetroProvider`.

File: sormas-app/app/src/main/java/de/symeda/sormas/app/lbds/LbdsDtoHelper.java [hunk 20]
```diff
@@ -0,0 +1,206 @@
...
+		CaseDataDto.DISEASE,
+		CaseDataDto.REPORT_DATE,
+		CaseDataDto.REPORTING_USER,
+		CaseDataDto.CASE_CLASSIFICATION,
+		CaseDataDto.INVESTIGATION_STATUS,
+		CaseDataDto.REGION,
+		CaseDataDto.DISTRICT,
+		CaseDataDto.FACILITY_TYPE,
+		CaseDataDto.HEALTH_FACILITY,
+		// Necessary for deserialization:
```
Comment (syntakker) [inline sormas-app/app/src/main/java/de/symeda/sormas/app/lbds/LbdsDtoHelper.java:75]: The following list of properties should be treated generically: All of these are `boolean` primitives, which is the reason for them being listed here.

File: sormas-app/app/build.gradle (hunk 12 above)
[the remark this answers is not in the document]
Comment (MartinWahnschaffe) [reply sormas-app/app/build.gradle:36]: Just add an entry to the developer documentation that this needs to be commented when it's necessary to have a debug version for LBDS.

File: sormas-app/app/src/main/java/de/symeda/sormas/app/lbds/LbdsDtoHelper.java (hunk 13 above)
[the remark this answers is not in the document]
Comment (syntakker) [reply sormas-app/app/src/main/java/de/symeda/sormas/app/lbds/LbdsDtoHelper.java:88]: Obsolete after treating primitives generically.

File: sormas-api/src/main/java/de/symeda/sormas/api/utils/pseudonymization/DtoPseudonymizer.java (hunk 4 above)
Comment (syntakker) [reply sormas-api/src/main/java/de/symeda/sormas/api/utils/pseudonymization/DtoPseudonymizer.java:110]: done

File: sormas-api/src/main/java/de/symeda/sormas/api/utils/pseudonymization/PseudonymizableDto.java (hunk 11 above)
[the remark this answers is not in the document]
Comment (syntakker) [reply sormas-api/src/main/java/de/symeda/sormas/api/utils/pseudonymization/PseudonymizableDto.java:23]: reverted, covered by generic treatment of primitives.

File: sormas-app/app/build.gradle (hunk 12 above)
[the remark this answers is not in the document]
Comment (syntakker) [reply sormas-app/app/build.gradle:36]: Reverted. The issue will be covered in the documentation on how to debug / trouble shooting.

File: sormas-app/app/src/main/java/de/symeda/sormas/app/LbdsRecevierComponent.java (hunk 5 above)
Comment (syntakker) [reply sormas-app/app/src/main/java/de/symeda/sormas/app/LbdsRecevierComponent.java:65]: done

File: sormas-app/app/src/main/java/de/symeda/sormas/app/LbdsRecevierComponent.java (hunk 6 above)
Comment (syntakker) [reply sormas-app/app/src/main/java/de/symeda/sormas/app/LbdsRecevierComponent.java:99]: done

File: sormas-app/app/src/main/java/de/symeda/sormas/app/LbdsRecevierComponent.java (hunk 8 above)
Comment (syntakker) [reply sormas-app/app/src/main/java/de/symeda/sormas/app/LbdsRecevierComponent.java:157]: Refactored to only send new cases / persons that have not been sent via LBDS yet.

File: sormas-app/app/src/main/java/de/symeda/sormas/app/LbdsRecevierComponent.java (hunk 9 above)
Comment (syntakker) [reply sormas-app/app/src/main/java/de/symeda/sormas/app/LbdsRecevierComponent.java:213]: See above.

File: sormas-app/app/src/main/java/de/symeda/sormas/app/backend/caze/CaseDtoHelper.java (hunk 14 above)
Comment (syntakker) [reply sormas-app/app/src/main/java/de/symeda/sormas/app/backend/caze/CaseDtoHelper.java:89]: done

File: sormas-app/app/src/main/java/de/symeda/sormas/app/backend/person/PersonDtoHelper.java (hunk 15 above)
Comment (syntakker) [reply sormas-app/app/src/main/java/de/symeda/sormas/app/backend/person/PersonDtoHelper.java:65]: done

File: sormas-app/app/src/main/java/de/symeda/sormas/app/lbds/LbdsDtoHelper.java (hunk 20 above)
Comment (syntakker) [reply sormas-app/app/src/main/java/de/symeda/sormas/app/lbds/LbdsDtoHelper.java:75]: done

File: sormas-app/app/src/main/java/de/symeda/sormas/app/lbds/LbdsDtoHelper.java (hunk 13 above)
[the remark this answers is not in the document]
Comment (syntakker) [reply sormas-app/app/src/main/java/de/symeda/sormas/app/lbds/LbdsDtoHelper.java:88]: done

File: sormas-app/app/src/main/java/de/symeda/sormas/app/lbds/LbdsDtoHelper.java (hunk 10 above)
Comment (syntakker) [reply sormas-app/app/src/main/java/de/symeda/sormas/app/lbds/LbdsDtoHelper.java:142]: Obsolete after refactoring. Will be removed.

File: sormas-app/app/src/main/java/de/symeda/sormas/app/lbds/LbdsIntentSender.java (hunk 17 above)
Comment (syntakker) [reply sormas-app/app/src/main/java/de/symeda/sormas/app/lbds/LbdsIntentSender.java:61]: done

File: sormas-app/app/src/main/java/de/symeda/sormas/app/lbds/LbdsIntentSender.java (hunk 16 above)
Comment (syntakker) [reply sormas-app/app/src/main/java/de/symeda/sormas/app/lbds/LbdsIntentSender.java:66]: done

File: sormas-app/app/src/main/java/de/symeda/sormas/app/lbds/LbdsIntentSender.java (hunk 18 above)
Comment (syntakker) [reply sormas-app/app/src/main/java/de/symeda/sormas/app/lbds/LbdsIntentSender.java:98]: done

File: sormas-app/app/src/main/java/de/symeda/sormas/app/lbds/LbdsIntentSender.java (hunk 19 above)
Comment (syntakker) [reply sormas-app/app/src/main/java/de/symeda/sormas/app/lbds/LbdsIntentSender.java:192]: done

File: sormas-app/app/src/main/java/de/symeda/sormas/app/settings/SettingsFragment.java (hunk 1 above)
Comment (syntakker) [reply sormas-app/app/src/main/java/de/symeda/sormas/app/settings/SettingsFragment.java:98]: Has been discussed: for the prototype phase we will keep two buttons for better control. Will be consolidated after.

File: sormas-rest/src/main/java/de/symeda/sormas/rest/PersonResource.java (hunk 3 above)
Comment (syntakker) [reply sormas-rest/src/main/java/de/symeda/sormas/rest/PersonResource.java:56]: done

File: sormas-app/keystore.properties.example (hunk 2 above)
Comment (syntakker) [reply sormas-app/keystore.properties.example:1]: done

File: sormas-app/app/src/main/java/de/symeda/sormas/app/LbdsRecevierComponent.java (hunk 7 above)
Comment (syntakker) [reply sormas-app/app/src/main/java/de/symeda/sormas/app/LbdsRecevierComponent.java:68]: done

**Comment by syntakker** [comment]
This PR has been replaced with a PR with the current status after rebasing on the current development branch: #6037

**Comment by syntakker** [comment]
Obsolete, see #6037
